salt/modules/datadog_api.py

```diff
@@ -262,3 +259,426 @@ def post_event(api_key=None,
+    :type name: str/list
+
+    :param group_states: Include additional information regarding group states
+    :type group_states: str
+
+    :param tags: A list of tags to filter by
+    :type tags: list
+
+    :param monitor_tags: A list of service or custom tags to filter by
+    :type monitor_tags: list
+
+    :param with_downtimes: Include current downtimes
+    :type with_downtimes: bool
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-call datadog.get_monitor id=4861948 \\
+            app_key='0123456789' \\
+            api_key='9876543210'
+
+    Note:
+
+    The tag filters act only on monitors specified by name.
+
+    Note:
+
+    If id is passed to this function, all other filter arguments will be ignored
+    '''
+
+    __utils__['datadog.initialize_connection'](api_key, app_key)
+
+    ret = {
+        'result': False,
+        'response': None,
+        'comment': ''
+    }
+
+    if __opts__['test']:
+        message = (
+            'A datadog monitor read call would be called for {}'
+        )
+        ret['result'] = None
+        ret['comment'] = message.format(
+            'ids {}'.format(id) if id is not None else 'names {}'.format(name)
+        )
+        return ret
+
+    response = []
+    if id is not None:
+        ids = []
+        if isinstance(id, list):
+            ids += id
+        else:
+            ids += [id]
+
+        for _id in ids:
+            if _id:
+                res = datadog.api.Monitor.get(
+                    _id,
+                    group_states=group_states
+                )
+                if 'errors' in res.keys():
+                    ret['response'] = res
+                    ret['comment'] = (
+                        'An error occured when trying to read monitors.'
+                    )
+                    return ret
+                else:
+                    response += [res]
+    else:
+        names = []
+        if isinstance(name, list):
+            names += name
+        else:
+            names += [name]
+
+        for _name in names:
+            res = __utils__['datadog.get_all_monitors'](
+                name=_name,
+                group_states=group_states,
+                tags=tags,
+                monitor_tags=monitor_tags,
+                with_downtimes=with_downtimes
+            )
+            if not res['result']:
+                ret['response'] = res['response']
+                ret['comment'] = (
+                    'An error occured when trying to read monitors.'
+                )
+                return ret
+            else:
+                response += res['response']
+
+    ret['result'] = True
+    ret['response'] = response
+    return ret
+# pylint: enable=dangerous-default-value
+
+
+# pylint: disable=dangerous-default-value
+def update_monitor(api_key=None,
+                   app_key=None,
+                   id=None,
+                   name=None,
+                   query=None,
+                   message='',
```

Comment:
Same comment here. :)

Instead of me commenting on all of the functions that do this, the functions in the state below that behave this way should also be addressed.